ui/src/app/lib/atlasmap-data-mapper/models/config.model.ts

```diff
@@ -197,9 +197,12 @@ export class ConfigModel {
     docDef.inspectionSource = docInitModel.inspectionSource;
     docDef.inspectionResult = docInitModel.inspectionResult;
     docDef.selectedRoot = docInitModel.selectedRoot;
-    docDef.uri = 'atlas:' + docDef.type.toLowerCase() + ':' + docDef.id;
-    if (docDef.type === DocumentType.JAVA) {
+    if (docDef.type === DocumentType.XSD) {
+      docDef.uri = 'atlas:' + 'xml' + ':' + docDef.id;
+    } else if (docDef.type === DocumentType.JAVA) {
       docDef.uri += '?className=' + docDef.inspectionSource;
```

Comment:
isn't it going to miss `atlas:java:id` prefix?

Reply:
So actually wondering if we should add `atlas:java:id` when `docDef.type === DocumentType.JAVA` **OR** `docDef.type === DocumentType.JAVA_ARCHIVE`. I'm not sure if it's possible at all to get `DocumentType.JAVA_ARCHIVE` passed in here.

Reply:
so - this is what we've got now:
    if (docDef.type === DocumentType.XSD) {
      docDef.uri = 'atlas:' + 'xml' + ':' + docDef.id;
    } else {
      docDef.uri = 'atlas:' + docDef.type.toLowerCase() + ':' + docDef.id;
    }

    if (docDef.type === DocumentType.JAVA) {
      docDef.uri += '?className=' + docDef.inspectionSource;
    }

So - is this what you are suggesting?
    if (docDef.type === DocumentType.XSD) {
      docDef.uri = 'atlas:' + 'xml' + ':' + docDef.id;
    } else {
      docDef.uri = 'atlas:' + docDef.type.toLowerCase() + ':' + docDef.id;
    }

    if (docDef.type === DocumentType.JAVA ||docDef.type === DocumentType.JAVA_ARCHIVE ) {
      docDef.uri += '?className=' + docDef.inspectionSource;
    }

Reply:
I would suggest this

```java
if (docDef.type === DocumentType.XSD) {
  docDef.uri = 'atlas:' + 'xml' + ':' + docDef.id;
} else if (docDef.type === DocumentType.JAVA_ARCHIVE) {
  docDef.uri = 'atlas:' + 'java' + ':' + docDef.id;
} else {
  docDef.uri = 'atlas:' + docDef.type.toLowerCase() + ':' + docDef.id;
}

if (docDef.type === DocumentType.JAVA ||docDef.type === DocumentType.JAVA_ARCHIVE ) {
  docDef.uri += '?className=' + docDef.inspectionSource;
}
```

might be even better splitting the enum into two, module type (java, json, xml) and others (schema, instance, etc) but maybe next time